tempy/widgets.py

```diff
@@ -181,11 +183,92 @@ def make_footer(self, footer):
         self._make_table_part('footer', footer)
 
     def make_caption(self, caption):
-        """Adds/Sobstitute the table's caption."""
+        """Adds/Substitutes the table's caption."""
         if not hasattr(self, 'caption'):
             self(caption=Caption())
         return self.caption.empty()(caption)
 
+    def col_class(self, css_class, col_index=None):
+        # adds css_class to every cell
+        if col_index is None:
+            for row in self.body.childs:
+                for cell in row.childs:
+                    cell.attr(klass=css_class)
+        else:
+            for row in self.body.childs:
+                if self.is_col_within_bounds(col_index, row):
+                    row.childs[col_index].attr(klass=css_class)
+
+    def row_class(self, css_class, row_index=None):
+        # adds css_class to every row
+        if row_index is None:
+            for row in self.body.childs:
+                row.attr(klass=css_class)
+        elif self.is_row_within_bounds(row_index):
+            self.body.childs[row_index].attr(klass=css_class)
+
+    def map_col(self, col_function, col_index=None):
+        # applies function to every cell
+        if col_index is None:
+            self.map_table(col_function)
+        else:
+            for row in self.body.childs:
+                if self.is_col_within_bounds(col_index, row):
+                    row.childs[col_index].childs[0] = col_function(row.childs[col_index].childs[0])
+
+    def map_row(self, row_function, row_index=None):
+        # applies function to every row
+        if row_index is None:
+            self.map_table(row_function)
+        elif self.is_row_within_bounds(row_index):
+            for cell in self.body.childs[row_index].childs:
+                if len(cell.childs) > 0:
+                    cell.childs[0] = row_function(cell.childs[0])
+
+    def map_table(self, format_function):
+        for row in self.body.childs:
+            for cell in row.childs:
+                if len(cell.childs) > 0:
+                    cell.childs[0] = format_function(cell.childs[0])
```

Comment:
I just noticed that in those methods the function is applied to only the first child in the cell. Isn't better to apply the function to every element inside the cell? Don't know, never thought about that.. what you think?

Is it safe enough to assume that the user will apply functions to coherent cells? i.e: map_table will be used on a table in which every single cell contains a valid input for the applied function?
In the end it is the user's concern to check for the input of his functions.. In that case a `cell.childs = map(format_function , cell.childs)` would work better.

Also, what if one or more of this function calls breaks? Would it be useful to add a `ignore_errors` parameter so every function call is performed in a `try.. except` and exceptions are raised conditionally?

What you think?

Reply:
For the first issue you've mentioned I agree if you are thinking of putting multiple tags inside a table cell or type different than a string

For the exceptions part, are you thinking about putting that parameter in each of these newly implemented methods?

Reply:
1) Yeah I was thinking the situation in which a cell contains something more than a single string (multiple strings, other tag(s)). A cell is a TemPy tag so is can contain a list of anything...

2) Yes, I think the ignore_errors parameter should be in every newly implemented map_* method. It's a sort of "graceful function application" for those situation in which a formatting is needed but if something goes wrong better have a bad string than nothing..

Reply:
Is there any method implemented that set's content's text? I only see the text() and render() methods. I'm just asking so I don't implement and make duplicated code.

Reply:
No there isn't, both render() and text() retrive the tag's content (with and without the html code). Insertion can be made with the APIs (text or anything) that in the end refers to DOMElement._insert(), there is no insertion specific for text though.

I thought about a string-specific method but that not seemed necessary, and yes seems a little tricky with nested ones..